corehq/apps/app_manager/suite_xml.py

```diff
@@ -1615,7 +1615,8 @@ def entries(self):
                     )
                 )
                 if isinstance(module, Module):
-                    for datum_meta in self.get_datum_meta_module(module, use_filter=False):
+                    use_filter = False if module.has_no_forms() else bool(module.case_details.short.filter)
```

Comment:
Oh this is correct, but the function name is wrong. `has_no_forms` should be `has_forms`